libwallet/assets/wallet/wallet_shared.go

```diff
@@ -129,11 +135,16 @@ func (wallet *Wallet) prepare() (err error) {
 	// operations and start go routine to listen for shutdown signal
 	wallet.cancelFuncs = make([]context.CancelFunc, 0)
 	wallet.shuttingDown = make(chan bool)
+	wallet.isCancelDone = make(chan struct{})
+
 	go func() {
 		<-wallet.shuttingDown
 		for _, cancel := range wallet.cancelFuncs {
 			cancel()
 		}
+
+		// Indicate completion of canceling all active contexts
+		wallet.isCancelDone <- struct{}{}
```

Comment:
when you add wallet.isCancelDone <- struct{}{} in here, that mean the Shutdown() function need wait wallet.isCancelDone to complete, so why you don't move bellow code to Shutdown() ?

```
for _, cancel := range wallet.cancelFuncs {
			cancel()
		}
```

Reply:
okay